File: src/http/Encoding.zig
pub const Encoding = enum {
    identity,
    gzip,
    deflate,
    brotli,
    zstd,
    chunked,

    pub fn canUseLibDeflate(this: Encoding) bool {
        return switch (this) {
            .gzip, .deflate => true,
            else => false,
        };
    }

    pub fn isCompressed(this: Encoding) bool {
        return switch (this) {
            .brotli, .gzip, .deflate, .zstd => true,
            else => false,
        };
    }

    /// Convert encoding to Content-Encoding header value
    pub fn toString(this: Encoding) []const u8 {
        return switch (this) {
            .brotli => "br",
            .gzip => "gzip",
            .zstd => "zstd",
            .deflate => "deflate",
            .identity => "identity",
            .chunked => unreachable, // chunked is Transfer-Encoding only
        };
    }
};

File: src/http_types/Encoding.rs
#[derive(Copy, Clone, Eq, PartialEq, Debug)]
pub enum Encoding {
    Identity,
    Gzip,
    Deflate,
    Brotli,
    Zstd,
    Chunked,
}

impl Encoding {
    /// Parses one case-insensitive coding token (RFC 9110 §8.4.1 / RFC 9112 §7); `None` if unrecognized.
    pub fn from_token(token: &[u8]) -> Option<Encoding> {
        use bun_core::strings::eql_case_insensitive_ascii_check_length as eql;
        if eql(token, b"gzip") || eql(token, b"x-gzip") {
            Some(Encoding::Gzip)
        } else if eql(token, b"deflate") {
            Some(Encoding::Deflate)
        } else if eql(token, b"br") {
            Some(Encoding::Brotli)
        } else if eql(token, b"zstd") {
            Some(Encoding::Zstd)
        } else if eql(token, b"identity") {
            Some(Encoding::Identity)
        } else if eql(token, b"chunked") {
            Some(Encoding::Chunked)
        } else {
            None
        }
    }

    pub fn can_use_lib_deflate(self) -> bool {
        matches!(self, Encoding::Gzip | Encoding::Deflate)
    }

    pub fn is_compressed(self) -> bool {
        matches!(
            self,
            Encoding::Brotli | Encoding::Gzip | Encoding::Deflate | Encoding::Zstd
        )
    }
}
